txscript/opcode.go

```diff
@@ -1944,6 +1943,44 @@ func opcodeHash256(op *opcode, data []byte, vm *Engine) error {
 	return nil
 }
 
+// opcodeCat concatenates the top two stack items, leaving the result on the
+// stack.
+//
+// Stack transformation: [...x1 x2] -> [... x1|x2]
+func opcodeCat(op *opcode, data []byte, vm *Engine) error {
+	// This op code can only be used if tapsript execution is active.
+	// Before the soft fork, this opcode was disabled.
+	if vm.taprootCtx == nil {
+		return opcodeDisabled(op, data, vm)
+	}
+
+	x2, err := vm.dstack.PopByteArray()
+	if err != nil {
+		return err
+	}
+
+	x1, err := vm.dstack.PopByteArray()
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
```

Comment:
Note to self: could preallocate known number of bytes before copying.

Reply:
Done: https://github.com/btcsuite/btcd/pull/2095/commits/5e4aee20946bb74aa42b10f80e4d6cd24d177680